### ISSUE TITLE
Gitlab Sentry-Backend DSN Support

### ISSUE THREAD

**Issue by AtjonTV** [getsentry/sentry-elixir#506]
Hello!

Today I got the urge to try Sentry-Elixir with the [Gitlab Sentry-backend](https://docs.gitlab.com/ee/operations/error_tracking.html#integrated-error-tracking), but sadly the way this client parses the DSN and creates the endpoint is not compatible with it.

I debugged a little and compared the source with how [the Ruby version does it](https://github.com/getsentry/sentry-ruby/blob/master/sentry-ruby/lib/sentry/dsn.rb).

From my debugging I have gathered the following diff:

```diff
diff --git a/lib/sentry/client.ex b/lib/sentry/client.ex
index 0a002ed..6dc84e6 100644
--- a/lib/sentry/client.ex
+++ b/lib/sentry/client.ex
@@ -257,9 +257,11 @@ defmodule Sentry.Client do
          %URI{userinfo: userinfo, host: host, port: port, path: path, scheme: protocol}
          when is_binary(path) and is_binary(userinfo) <- URI.parse(dsn),
          [public_key, secret_key] <- keys_from_userinfo(userinfo),
-         [_, binary_project_id] <- String.split(path, "/"),
+         uri_path <- String.split(path, "/"),
+         {binary_project_id, uri_path} <- List.pop_at(uri_path, -1),
+         base_path <- Enum.join(uri_path, "/"),
          {project_id, ""} <- Integer.parse(binary_project_id),
-         endpoint <- "#{protocol}://#{host}:#{port}/api/#{project_id}/envelope/" do
+         endpoint <- "#{protocol}://#{host}:#{port}#{base_path}/api/#{project_id}/store/" do
       {endpoint, public_key, secret_key}
     else
       _ ->
```

With this diff applied the Gitlab Sentry-backend DSN is parsed correctly, and the endpoint url is also constructed correctly.

```
iex(27)> Debug.get_dsn_debug "https://glet_asdasdasd@gitlab.atvg-studios.atvg.cloud/api/v4/error_tracking/collector/276"          
{"https://gitlab.atvg-studios.atvg.cloud:443/api/v4/error_tracking/collector/api/276/store/", "glet_asdasdasd", nil}
```

As I am new to Elixir I am not sure if this is the best possible implementation, because of this I am creating this issue instead of a merge request.

I would love to see this fixed in the client and am willing to open a MR if I get feedback on this. Alternatively a maintainer could also just quickly implement that patch.

**Comment by AtjonTV** [comment]
While this patch works to allow Gitlab DSNs in Sentry-Elixir, sadly Gitlab doesnt seem to support Enveloped Events. (which is a Gitlab issue)

Still, this would bring the way the DSN is parsed and Endpoint constructed to the same level as in the Ruby implementation.

**Comment by mitchellhenke** [comment]
A PR/MR would be most welcome!

**Comment by AtjonTV** [comment]
I have fixed my mess up of changing `/envelope` to `/store` and opend a MR: #507

**Comment by mitchellhenke** [comment]
Thank you!!

Closed by #507
